### ISSUE TITLE
SwiftLint doesn't complain when rule is configured twice

### ISSUE THREAD

**Issue by Fryie** [realm/SwiftLint#2191]
### New Issue Checklist

- [x] Updated SwiftLint to the latest version
- [x] I searched for [existing GitHub issues](https://github.com/realm/SwiftLint/issues)

### Bug Report
Steps to reproduce:
1. Create a new project (e.g. `swift package init`)
2. Add a `.swiftlint.yml` with the following contents:
```yaml
identifier_name:
  severity: error
identifier_name:
  severity: warning
```
3. Run `swiftlint`

Expected outcome: SwiftLint complains that the rule is configured twice (in this case, even in an incompatible way)
Effective outcome: No warning is raised and apparently the first configuration is used (not sure if this is always the case).
Why this could be a problem: I encountered a duplicate rule configuration in one of our `.swiftlint.yml`. This had gone unnoticed for quite a while, so the rule didn't do what it was supposed to.

Interestingly, if either of the rule configurations are invalid (e.g. `identifier_name: foo`), SwiftLint will complain, suggesting that both configurations are actually being parsed.

### Environment

* SwiftLint version

I tested this both on the released v0.25.1 and the current master

* Installation method used (Homebrew, CocoaPods, building from source, etc)?

Homebrew for the released version and from source for the current master

* Paste your configuration file:

see above

* Are you using [nested configurations](https://github.com/realm/SwiftLint#nested-configurations)?

no

* Which Xcode version are you using (check `xcode-select -p`)?

This just returns `/Applications/Xcode.app/Contents/Developer` for me.
